File: docker_compose/feinplanung/HiCuMes.Persistence/Entities/CdMachine.cs
using System;
using System.Collections.Generic;

namespace HiCuMes.Persistence.Entities;

public partial class CdMachine
{
    public long Id { get; set; }

    public DateTime? CreateDateTime { get; set; }

    public string? ExternalId { get; set; }

    public DateTime? UpdateDateTime { get; set; }

    public string? VersionNr { get; set; }

    public string? Name { get; set; }

    public long? MachineTypeId { get; set; }

    public  CdMachinetype? MachineType { get; set; }

    public  ICollection<MachineOccupation> Machineoccupations { get; set; } = new List<MachineOccupation>();

    public  ICollection<Machinestatushistory> Machinestatushistories { get; set; } = new List<Machinestatushistory>();

    public  ICollection<Toolsetting> Toolsettings { get; set; } = new List<Toolsetting>();
}

File: core/entities/src/main/java/de/iisys/sysint/hicumes/core/entities/CD_Machine.java
package de.iisys.sysint.hicumes.core.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonView;
import de.iisys.sysint.hicumes.core.entities.jsonViews.JsonViews;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@ToString(callSuper = true)
@XmlRootElement(name = "cD_Machine")
@XmlAccessorType(XmlAccessType.FIELD)
@Cacheable
@Table(indexes = @Index(columnList = "externalId"))
@org.hibernate.annotations.Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class CD_Machine extends EntitySuperClass {

    @ManyToOne
    @JsonView({JsonViews.ViewMachineOccupation.class, JsonViews.SchemaMapperDefault.class})
    @ToString.Exclude
    private CD_MachineType machineType;

    private String name;

    @JsonView(JsonViews.SchemaMapperDefault.class)
    @OneToMany(mappedBy = "machine", cascade = CascadeType.ALL)
    private List<MachineOccupation> machineOccupations = new ArrayList<>();

    @JsonView(JsonViews.SchemaMapperDefault.class)
    @OneToMany(mappedBy = "machine", cascade = CascadeType.ALL)
    private List<MachineStatusHistory> machineStatusHistories = new ArrayList<>();

    @JsonView(JsonViews.SchemaMapperDefault.class)
    @OneToMany(mappedBy = "machine", cascade = CascadeType.ALL)
    @ToString.Exclude
    private List<ToolSetting> toolSettings = new ArrayList<>();


    public CD_Machine(String name, CD_MachineType machineType) {
        machineType.machinesAdd(this);
        this.name = name;
        this.machineType = machineType;
    }

    @JsonIgnore
    public void machineOccupationsAdd(MachineOccupation input) {
        input.setMachine(this);
        machineOccupations.add(input);
    }

    @JsonIgnore
    public void machineStatusHistoriesAdd(MachineStatusHistory input) {
        input.setMachine(this);
        machineStatusHistories.add(input);
    }

    @JsonIgnore
    public void toolSettingsAdd(ToolSetting input) {
        input.setMachine(this);
        toolSettings.add(input);
    }


}
